yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -39,6 +39,14 @@ import org.apache.spark.deploy.history.HistoryServer
 import org.apache.spark.scheduler.cluster.YarnSchedulerBackend
 import org.apache.spark.scheduler.cluster.CoarseGrainedClusterMessages._
 import org.apache.spark.util.{AkkaUtils, SignalLogger, Utils}
+import org.apache.hadoop.mapreduce.security.TokenCache
+import org.apache.hadoop.io.Text
+import org.apache.hadoop.security.UserGroupInformation
+import javax.crypto.{SecretKey, KeyGenerator}
+import org.apache.hadoop.mapreduce.CryptoUtils
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.mapreduce.MRJobConfig
+import java.security.NoSuchAlgorithmException
```

Comment:
These imports need to be re-ordered according to Spark's code style guide: https://cwiki.apache.org/confluence/display/SPARK/Spark+Code+Style+Guide